[IMP] sales: updated amazon order management (#6363)
odoo/documentation: content/applications/sales/sales/amazon_connector/manage.rst

=======================
Amazon order management
=======================

Order synchronization
=====================

Orders are automatically fetched from Amazon, and synchronized in Odoo, at regular intervals.

The synchronization is based on the Amazon status: only orders whose status has changed since the
last synchronization are fetched from Amazon.

For *FBA* (Fulfilled by Amazon), only *Shipped* and *Canceled* orders are fetched.

For *FBM* (Fulfilled by Merchant), the same is done for *Unshipped* and *Canceled* orders. For each
synchronized order, a sales order and customer are created in Odoo (if the customer is not already
registered in the database).

.. note::
   When an order is canceled in Amazon, and was already synchronized in Odoo, the corresponding
   sales order is automatically canceled in Odoo.

Force synchronization
=====================

> In order to force the synchronization of an order, whose status has **not** changed since the

Reviewer: not changed...in Odoo or Amazon?
It's important to be specific here if we're going to evoke **bold** emphasis like that

previous synchronization, start by activating the :ref:`developer mode <developer-mode>`.

Then, navigate to the Amazon account in Odoo (:menuselection:`Sales app --> Configuration -->
Settings --> Connectors --> Amazon Sync --> Amazon Accounts`), and modify the date under
:menuselection:`Orders Follow-up --> Last Order Sync`.

> Be sure to pick a date that occurs prior to the last status change of the desired order to

Reviewer: Why would we sync to a *prior* date?

Reviewer 2: This is the only way a force sync will work. 
 
Added in clarification.

synchronize and save.

.. tip::
   To immediately synchronize the orders of an Amazon account, switch to :ref:`developer mode
   <developer-mode>`, head to the Amazon account in Odoo, and click :guilabel:`Sync Orders`. The
   same can be done with pickings by clicking :guilabel:`Sync Pickings`.

Manage deliveries in FBM
========================

Whenever an FBM (Fulfilled by Merchant) order is synchronized in Odoo, a picking is instantly

> created in the *Inventory* app, along with a sales order and customer. Then, decide to either ship

Reviewer: ```suggestion
created in the *Inventory* app, along with a sales order and customer record. Then, decide to either ship
```
OR
```suggestion
created in the *Inventory* app, along with a sales order and *Contacts* record for the customer. Then, decide to either ship
```

Reviewer 2: Went with A

all the ordered products to the customer at once, or ship products partially using backorders.

When a picking related to the order is confirmed, a notification is then sent to Amazon, who, in
turn, notifies the customer that the order (or a part of it) is on its way.

.. important::
   Amazon requires users to provide a tracking reference with each delivery. This is needed to
   assign a carrier.

   If the carrier doesn't automatically provide a tracking reference, one must be set manually. This

>    rule applies to all marketplaces.

Reviewer: ```suggestion
   rule applies to all marketplaces in Odoo.
```

Reviewer 2: This refers to Amazon marketplaces not Odoo

Reviewer 2: Added in clarification.

.. tip::
   - If the chosen carrier isn't supported by Odoo, a carrier with the same name can still be

>      created (e.g. create a carrier named `Colissimo`). The name used is **not** case sensitive, but

Reviewer: This example is too localized and would confuse anyone not in the EU.
Would change to `DHL` or `FedEx` maybe for more universal understanding.

Reviewer 2: We support both DHL and Fedex and this may cause some folks to think otherwise. I'm going to say EasyShip as a neutral example of a carrier not supported by Odoo.

     be mindful to avoid typos. If there are typos, Amazon will **not** recognize them.

>    - Next, create a delivery carrier named `Self Delivery` to inform Amazon that the user will make

Reviewer: I don't understand this **Tip**.
The use case isn't clear to me here with the instructions provided.
Is a specific carrier delivering a parcel or is the user?

Reviewer 2: This is for a unsupported carrier (by Odoo)

Reviewer 2: I've removed the bullet points.

     the deliveries. Even with this route, a tracking reference still **must** be entered.
   - Remember, the customer is notified by email about the delivery, and the carrier, along with the
     tracking reference, are displayed in the email to the customer.

.. seealso::

>    - :doc:`/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper`

Reviewer: Use the relative link

.. warning::
   TaxCloud integration will be decommissioned in Odoo 17+.

Register payments
-----------------

Since customers pay Amazon as an intermediary, creating a dedicated *Bank* journal (e.g. named
`Amazon Payments`), with a dedicated *Bank and Cash* intermediary account is recommended.

Additionally, as Amazon makes a single monthly payment, selecting all the invoices linked to a
single payment is necessary when registering payments.

To do that, use the appropriate :guilabel:`Journal` dedicated to Amazon payments, and select
:guilabel:`Batch Deposit` as the :guilabel:`Payment Method`.

Then, select all the generated payments, and click :menuselection:`Actions --> Create batch payment
--> Validate`.

.. tip::
   This same action can be performed with vendor bills from Amazon dedicated to commissions.

   When the balance is received in the bank account at the end of the month, and the bank statements
   are recorded, credit the Amazon intermediary account by the amount received.

Follow Amazon sales in sales reporting
======================================

On the Amazon account profile in Odoo , a sales team is set under the :guilabel:`Order Follow-up`
tab.

> This gives quick access to important metrics related to sales reporting. By default, the Amazon account's

Reviewer: Line is too long.

=======================
Amazon order management
=======================

Order synchronization
=====================

Orders are automatically fetched from Amazon, and synchronized in Odoo, at regular intervals.

The synchronization is based on the Amazon status: only orders whose status has changed since the
last synchronization are fetched from Amazon.

For *FBA* (Fulfilled by Amazon), only *Shipped* and *Canceled* orders are fetched.

For *FBM* (Fulfilled by Merchant), the same is done for *Unshipped* and *Canceled* orders. For each
synchronization order, a sales order and customer are created in Odoo, if they aren't already
registered.

.. note::
   When an order is canceled in Amazon, and was already synchronized in Odoo, the corresponding
   sales order is automatically canceled in Odoo.

> Force synchronization

Reviewer 3: FYI in Odoo 17 this section can change thanks to this feature: https://www.odoo.com/web#id=2806551&cids=1&menu_id=4720&action=333&active_id=4106&model=project.task&view_type=form

But we can keep a word if the user want to force the fetch of all changes since a specific date

Author: Okay. Good to know about 17 -- I'll be sure to keep that in mind. Thanks for letting me know. But, since this is targeting 16, can we leave this section as-is? @dalaOdoo

Reviewer 3: it was just an FYI for 17 so yes

=====================

> In order to force the synchronizationof an order, whose status has **not** changed since the

Reviewer 3: ```suggestion
In order to force the synchronization of an order, whose status has **not** changed since the
```

previous synchronization, start by activating the :ref:`developer mode <developer-mode>`.

> Then, navigate to the Amazon account, and modify the date under :menuselection:`Orders Follow-up

Author: Great point. Given the context, I'd assume this would occur in Odoo, because being in developer mode was the first step. So, I'll add a couple of clarifying mentions that it takes place in Odoo, so as to avoid any confusion. Thanks for bringing this to my attention. Again, my apologies for the lack of total clarity with these amazon docs -- I had to do my best to create them without any access to a paid-for Amazon Seller account.

Follow deliveries in FBA
========================

When an *FBA* (Fulfilled by Amazon) order is synchronized in Odoo, a stock move is recorded for each
sales order item. That way, it's saved in the system.

Inventory managers can access these stock moves by navigating to :menuselection:`Inventory app -->
Reporting --> Moves History`.

These managers pick up products in a specific inventory location called, *Amazon*. This specific
location represents stock in Amazon's warehouses, and allows users to manage the stock of products

> under the *FBA* program.

Reviewer 3: @ksc-odoo I don't understand what you are trying to say here.
For FBA orders, the stock move is made automatically by the connector thanks to the shipping status of Amazon.

When sending new products to amazon, the user should create a picking (delivery order) to transfer these products from his warehouse to the amazon location.

Author: Okay. I replaced this unclear section with the suggestion you left. Feel free to give it another look to see if everything is making sense now. Again, I apologize for any "confusing" portions of these Amazon connector docs. Without proper access to a valid Amazon Marketplace account, I am unable to have hands-on experience working with it, and can only do my best to enhance the previously written documentation about it, which is why I'm hoping your brilliant expertise can help "fill in the gaps" and make things clear and accurate. Any and all help/guidance you can give me in this regard would be greatly appreciated. Thanks! @dalaOdoo

.. tip::
   To follow *Amazon (FBA)* stock in Odoo, make an inventory adjustment after replenishing it. An
   automated replenishment from reordering rules can also be triggered on the Amazon location.

The Amazon location is configurable by accessing the Amazon account managed in Odoo. All accounts of
the same company use the same location, by default. However, it is possible to follow the stock
filtered by marketplace.

To do that, first remove the marketplace, where the desired stock to follow separately can be found,
from the list of synchronized marketplaces.

Next, create another registration for this account, and remove all marketplaces -- *except* the
marketplace this is desired to be isolated from the others.

Lastly, assign another stock location to the second registration of the account.

Invoices and register payments
==============================

Issue invoices
--------------

Due to Amazon's policy of not sharing customer email addresses, it is *not* possible to send
invoices directly to Amazon customers from Odoo. However, it *is* possible to manually upload the

> generated invoices from odoo to the Amazon back-end.

Reviewer 4: ```suggestion
generated invoices from Odoo to the Amazon back-end.
```

Additionally, for B2B clients, it is currently required to manually retrive VAT numbers from the

> Amazon back-end *before* creating an invoice in Odoo.

Reviewer 4: ```suggestion
Amazon back-end **before** creating an invoice in Odoo.
```

.. note::
   For :doc:`TaxCloud <../../../finance/accounting/taxes/taxcloud>` users: invoices created from
   Amazon sales orders are **not** synchronized with TaxCloud, since Amazon already includes them in
   its own tax report to TaxCloud.

.. warning::
   TaxCloud integration will be decommissioned in Odoo 17+.

Register payments
-----------------

Since customers pay Amazon as an intermediary, creating a dedicated *Bank* journal (e.g. named
`Amazon Payments`), with a dedicated *Bank and Cash* intermediary account is recommended.

Additionally, as Amazon makes a single monthly payment, selecting all the invoices linked to a
single payment is necessary when registering payments.

> To do that, used the appropriate :guilabel:`Journal` dedicated to Amazon payments, and select

Reviewer 4: ```suggestion
To do that, use the appropriate :guilabel:`Journal` dedicated to Amazon payments, and select
```

:guilabel:`Batch Deposit` as the :guilabel:`Payment Method`.

Then, select all the generated payments, and click :menuselection:`Actions --> Create batch payment
--> Validate`.

.. tip::
   This same action can be performed with vendor bills from Amazon dedicated to commissions.

   When the balance is received in the bank account at the end of the month, and the bank statements
   are recorded, credit the Amazon intermediary account by the amount received.

Follow Amazon sales in sales reporting
======================================

> On the Amazon account, a sales team is set under the :guilabel:`Order Follow-up` tab.

Reviewer 4: Can we also add where to find the Amazon account in Odoo?
```suggestion
On the Amazon account profile in Odoo , a sales team is set under the :guilabel:`Order Follow-up` tab.
```

> This gives quick access to important metrics related to Sales reporting. By default, the account's
> sales team is shared between all the company's accounts.

Reviewer 4: ```suggestion
This gives quick access to important metrics related to sales reporting. By default, the Amazon account's
sales team is shared between all of the company's accounts.
```

If desired, the sales team on the account can be changed for another, in order to perform a separate
reporting for the sales of this account.

.. tip::

>    It is also possible to perform reporting on a per-marketplace basis in a similar fashion.

Reviewer 4: ```suggestion
   It is also possible to perform reporting on a per-marketplace basis.
```

>    First, remove the desired marketplace from the list of synchronized marketplaces.

Reviewer 4: Again, can we include where to find the list of synchronized marketplaces in Odoo?

>    Then, create another registration for this account, and remove all other marketplaces, *except*

Reviewer 4: ```suggestion
   Then, create another registration for this account, and remove all other marketplaces **except**
```
